modules/tide_workflow_notification/tide_workflow_notification.module

```diff
@@ -0,0 +1,238 @@
+      '#rows' => 10,
+      '#default_value' => $notification_values[$transition]['message'] ?? $transition_settings['message'] ?? '',
+    ];
+
+    $form['notifications'][$transition]['token_tree'] = [
+      '#theme' => 'token_tree_link',
+      '#token_types' => ['node', 'user', 'workflow-notification:recipient'],
+      '#show_restricted' => TRUE,
+      '#weight' => 90,
+    ];
+  }
+
+  $form['actions']['submit']['#submit'][] = 'tide_workflow_notification_form_workflow_edit_form_submit';
+}
+
+/**
+ * Submit callback.
+ *
+ * @param array $form
+ *   Form.
+ * @param \Drupal\Core\Form\FormStateInterface $form_state
+ *   Form state.
+ */
+function tide_workflow_notification_form_workflow_edit_form_submit(array $form, FormStateInterface $form_state) {
+  $settings = \Drupal::configFactory()->getEditable('tide_workflow_notification.settings');
+  $notification_values = $form_state->getValue('notifications', []);
+  foreach ($notification_values as $transition => $value) {
+    $settings->set('notifications.' . $transition, $value);
+  }
+  $settings->save();
+}
+
+/**
+ * Implements hook_ENTITY_TYPE_presave().
+ */
+function tide_workflow_notification_node_presave(NodeInterface $node) {
+  try {
+    if ($node->isNew() || !$node->hasField('moderation_state') || empty($node->getLoadedRevisionId())) {
+      return;
+    }
+
+    $settings = \Drupal::config('tide_workflow_notification.settings');
+    $moderation_state = $node->get('moderation_state')->getString();
+
+    // Load the last revision as $node->original is the published one.
+    /** @var \Drupal\node\NodeInterface $last_revision */
+    $last_revision = \Drupal::entityTypeManager()
+      ->getStorage('node')
+      ->loadRevision($node->getLoadedRevisionId());
+    $last_moderation_state = $last_revision->get('moderation_state')
+      ->getString();
+
+    $transition = $last_moderation_state . '_' . $moderation_state;
+    $transition_settings = $settings->get('notifications.' . $transition) ?? [];
+    // Bail out early if the transition is not enabled for notification.
+    if (empty($transition_settings['enabled']) || empty($transition_settings['subject']) || empty($transition_settings['message'])) {
+      return;
+    }
+
+    $last_author = $last_revision->getRevisionUser();
+    $recipients = [];
+    switch ($transition) {
+      // Only notify the last author.
+      case 'needs_review_draft':
+        $recipients[$last_author->id()] = $last_author;
+        break;
+
+      // Notify all Approvers.
+      case 'draft_needs_review':
+        $recipients = _tide_workflow_notification_get_node_approvers($node, $transition);
+        break;
+
+      case 'draft_published':
+        $recipients = _tide_workflow_notification_get_node_approvers($node, $transition);
+        $recipients[$last_author->id()] = $last_author;
+        break;
+
+      // Notify all Approvers and the last author.
+      case 'needs_review_published':
+      case 'published_archived':
```

Comment:
```suggestion
      // Notify all Approvers and the last author.
      case 'needs_review_published':
      case 'published_archived':
      case 'draft_published':

```

Could the switch statement just include `draft_published` here?

Reply:
Hi @anthony-malkoun , fixed. thanks